SDL/Kernels.h

```diff
@@ -12,8 +12,9 @@
 
 namespace SDL {
   ALPAKA_FN_ACC ALPAKA_FN_INLINE void rmQuintupletFromMemory(struct SDL::quintuplets& quintupletsInGPU,
-                                                             unsigned int quintupletIndex) {
-    quintupletsInGPU.isDup[quintupletIndex] = true;
+                                                             unsigned int quintupletIndex,
+                                                             bool secondpass = false) {
+    quintupletsInGPU.isDup[quintupletIndex] |= 1 + secondpass;
```

Comment:
What is the reason for this change?

Reply:
This is related to the change to `char`. The reason for this is that there are two passes for duplicate cleaning, first in `removeDupQuintupletsInGPUAfterBuild` and then in `removeDupQuintupletsInGPUBeforeTC`. Previously, `removeDupQuintupletsInGPUBeforeTC` skipped over quintuplets that were already marked as duplicated to save time, but there was no way to differentiate whether it was marked as duplicate in the first or second pass, which is what gives rise to the race condition. In #387 I removed skipping over quintuplets marked as duplicate so that it's always consistent, but this way it ends up doing many more comparisons, which is why it slowed down a lot. So now, what I did was change `isDup` to a char so that we can use the lowest two bits to mark duplicates in the first and second pass. So we can still skip all quintuplets that were marked as duplicates in the first pass, while keeping things consistent. This is the same thing I did in #350.

Reply:
I should also mention that we're not taking up more memory after this change since both `bool` and `char` use 1 byte

Reply:
Got it, thank you! Merging.